File: android/src/main/kotlin/jyahann/dgis_map_kit/CameraController.kt
package jyahann.dgis_map_kit

import ru.dgis.sdk.map.Map
import io.flutter.plugin.common.MethodChannel
import ru.dgis.sdk.Duration
import ru.dgis.sdk.map.CameraAnimationType

class CameraController(map: Map, methodChannel: MethodChannel) {
    private var map: Map
    private var methodChannel: MethodChannel

    init {
        this.map = map
        this.methodChannel = methodChannel
    }

    fun moveCamera(params: kotlin.collections.Map<String, Any>) : Unit {
        map.camera.move(
            CameraUtils.getCameraPositionFromDart(params["cameraPosition"] as kotlin.collections.Map<String, Any>),
            Duration.ofMilliseconds((params["durationInMilliseconds"] as Int).toLong()) ,
            CameraAnimationType.valueOf(params["animationType"] as String)
        )
    }
}

File: ios/Classes/CameraController.swift
import DGis;
import Flutter;

class CameraController {
    private var map: DGis.Map;
    private var methodChannel: FlutterMethodChannel;
    private var cameraMoveCancellable: Future<CameraAnimatedMoveResult>?;
    
    init(map: DGis.Map, methodChannel: FlutterMethodChannel) {
        self.map = map;
        self.methodChannel = methodChannel;
    }
    
    func moveCamera(cameraPosition position: Dictionary<String, Any>) {
        self.cameraMoveCancellable = map.camera.move(
            position: CameraUtils.getCameraPositionFromDart(
                cameraPosition: position["cameraPosition"] as! Dictionary<String, Any>
            ),
            time: TimeInterval(floatLiteral: (position["durationInMilliseconds"] as! Double) * 0.001)
        )
    }
}
